woocommerce/wc-admin
# Implement inbox note read state

Fixes #7793

This PR implements `read` state for notes.

- Added `is_read` column to the `wp_wc_admin_notes` table.
- Note API returns `is_read`
- Removed `lastRead` 
- Make CTA buttons rectangle if there are more than 1 buttons (e.g, Yes or No buttons)
- Read state 

### Screenshots

**Multiple CTA buttons:**

![Screen Shot 2021-11-03 at 9 07 27 PM](https://user-images.githubusercontent.com/4723145/140256634-31b5da6c-a8b1-47e9-a86d-f4b52e00c444.jpg)

**Read vs Unread**
![Screen Shot 2021-11-03 at 9 11 05 PM](https://user-images.githubusercontent.com/4723145/140256696-64278cbb-969d-4db6-9ac0-784f79dd7f27.jpg)



### Detailed test instructions:

**Testing is_read column**
1. Lower `woocommerce_admin_version` value and navigate WooCommerce -> Home to update the table schema.
2. Confirm `wp_wc_admin_notes` table has `is_read` column

**Testing is_read value**
1. Navigate to WooCommerce -> Home
2. Click CTA button from one of the notes.
3. Notice the note CSS changes as soon as you click the button.
4. Confirm the `is_read` value of the note has been updated to `1`
